harness/determined/common/api/errors.py

```diff
@@ -38,7 +38,9 @@ def __init__(self, response: requests.Response) -> None:
 
 
 class NotFoundException(APIException):
```

Comment:
this is a change in behavior between what the constructor does now vs what it was doing before which comes with some risk. can we avoid it? Have you checked the usages of it? 
I see you're setting the same fields as the parent's parent so we should be fine.

Reply:
There are two places that assign `NotFoundException` -- in `_get_error_str` and `not_found_errs`, both of which should have the status code 404 anyway, but I had to change the constructor behavior to be able to modify the message.